### PR TITLE
Smq

### PR DESCRIPTION
**Pull request by s1752729916**

### CODE REVIEW

**Comment by qhanson** [comment]
Hi, do you get stable results? I train the model with the available wav files and the output is not very good.

**Comment by s1752729916** [comment]
> Hi, do you get stable results? I train the model with the available wav files and the output is not very good.

you mean the torch model? tested only on dataset5 and its performance is similar to tf model

**Comment by qhanson** [comment]
Can your share the loss cuve or pre-trained model? I train it from scrtach with the avaiable wav and bs-values. But I get unsatisfying results. I am still debugging this.

**Comment by SZNero** [comment]
use the dataset from dataset4 to dataset16 to train the model

**Comment by qhanson** [comment]
> use the dataset from dataset4 to dataset16 to train the model

What is dataset4 and dataset16? I only use the wav file and the correspoding bs_values to train the model from scratch, as follows

[bs_value_1114_1_01.npy](https://github.com/FACEGOOD/FACEGOOD-Audio2Face/blob/main/code/train/bs_value/bs_value_1114_1_01.npy)

[bs_value_1114_1_02.npy](https://github.com/FACEGOOD/FACEGOOD-Audio2Face/blob/main/code/train/bs_value/bs_value_1114_1_02.npy)

[bs_value_1114_2_01.npy](https://github.com/FACEGOOD/FACEGOOD-Audio2Face/blob/main/code/train/bs_value/bs_value_1114_2_01.npy)

[bs_value_1114_2_02.npy](https://github.com/FACEGOOD/FACEGOOD-Audio2Face/blob/main/code/train/bs_value/bs_value_1114_2_02.npy)

[bs_value_1114_2_03.npy](https://github.com/FACEGOOD/FACEGOOD-Audio2Face/blob/main/code/train/bs_value/bs_value_1114_2_03.npy)

[bs_value_1114_2_04.npy](https://github.com/FACEGOOD/FACEGOOD-Audio2Face/blob/main/code/train/bs_value/bs_value_1114_2_04.npy)

[bs_value_1114_2_05.npy](https://github.com/FACEGOOD/FACEGOOD-Audio2Face/blob/main/code/train/bs_value/bs_value_1114_2_05.npy)

[bs_value_1114_3_15.npy](https://github.com/FACEGOOD/FACEGOOD-Audio2Face/blob/main/code/train/bs_value/bs_value_1114_3_15.npy)

[bs_value_1114_3_16.npy](https://github.com/FACEGOOD/FACEGOOD-Audio2Face/blob/main/code/train/bs_value/bs_value_1114_3_16.npy)

**Comment by qhanson** [comment]
@s1752729916 hi, `torch.split()` may not behave as what we want.
```
split_y = torch.split(y,2,0)
split_y_ = torch.split(y_,2,0)
y0 = split_y[0]
y1 = split_y[1]
y_0 = split_y_[0]
y_1 = split_y_[1]
```
if the frame is 1, 2, 3, 4, 5, 6, 7, 8,
We wish to seperate the output to 1,3,5,7, and 2,4,6,8. However, `torch.split()` does not do this.

**Comment by s1752729916** [comment]
> @s1752729916 hi, `torch.split()` may not behave as what we want.
> 
> ```
> split_y = torch.split(y,2,0)
> split_y_ = torch.split(y_,2,0)
> y0 = split_y[0]
> y1 = split_y[1]
> y_0 = split_y_[0]
> y_1 = split_y_[1]
> ```
> 
> if the frame is 1, 2, 3, 4, 5, 6, 7, 8, We wish to seperate the output to 1,3,5,7, and 2,4,6,8. However, `torch.split()` does not do this.
It seems that you are right. This problem also exists in the TF implementation. The TF implementation even shuffles the input data. But the performance of tf training seems to be good, does it indicate that this loss term is insignificant?

**Comment by qhanson** [comment]
I agree that this loss term is insignificant compared to the P loss ( MSE between predicted blendshape and ground truth).
